test/system/255-auto-update.bats

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# Tests for automatically update images for containerized services
+#
+
+load helpers
+
+UNIT_DIR="/usr/lib/systemd/system"
+
+function setup() {
+    skip_if_remote "systemd tests are meaningless over remote"
+    skip_if_rootless
+
+    basic_setup
+}
+
+function generate_service() {
+    fake_img=$1
+    target_img=$2
+    cmd=$3
+    label=$4
+
+    cname=$(random_string)
+    echo $cname
+    run_podman tag $fake_img $target_img
+    if [ "$4" ]; then
+        run_podman run -d --name $cname --label "io.containers.autoupdate=$label" $target_img $cmd
+    else
+        run_podman run -d --name $cname $target_img $cmd
+    fi
+
+    run_podman generate systemd --new $cname
+    echo "$output" > "$UNIT_DIR/container-$cname.service"
+    run_podman rm -f $cname
+
+    systemctl daemon-reload
+
+    run systemctl start container-$cname
+    if [ $status -ne 0 ]; then
+        die "Error starting systemd unit $SERVICE_NAME, output: $output"
+    fi
+
+    run systemctl status container-$cname
+    if [ $status -ne 0 ]; then
+        die "Non-zero status of systemd unit $SERVICE_NAME, output: $output"
+    fi
+}
+
+@test "podman auto-update - label io.containers.autoupdate=image" {
+    run generate_service $IMAGE quay.io/libpod/alpine:latest top image
+    cname=${lines[0]}
+    echo $output
+    run_podman inspect --format "{{.Image}}" $cname
+    ori_image=$output
+
+    sleep 2
+    run_podman ps -a
+    run_podman auto-update
+    is "$output" "Trying to pull.*" "Image is updated."
+    run_podman ps -a
+    sleep 2
+    run_podman inspect --format "{{.Image}}" $cname
+    systemctl stop container-$cname
+    rm -f $UNIT_DIR/container-$cname.service
+    [[ "$output" != "$ori_image" ]]
+}
+
+@test "podman auto-update - label io.containers.autoupdate=disabled" {
+    run generate_service $IMAGE quay.io/libpod/alpine:latest top disabled
+    cname=${lines[0]}
+    echo $output
+    run_podman inspect --format "{{.Image}}" $cname
+    ori_image=$output
+
+    sleep 2
+    run_podman ps -a
+    run_podman auto-update
+    is "$output" "" "Image is not updated with disabled."
+    run_podman ps -a
+    sleep 2
+    run_podman inspect --format "{{.Image}}" $cname
+    systemctl stop container-$cname
+    rm -f $UNIT_DIR/container-$cname.service
+    is "$output" "$ori_image" "Image hash should not changed."
```

Comment:
Please assign a temporary named variable after line 81 to hold `$output`. Testing `$output` so far after `run` is going to cause confusion and possibly unrelated errors if a future maintainer adds an unrelated `run` command in between. Same applies to various other places in the code. 

Even better, create a `teardown()` function and do such cleanup there.

Reply:
OK. May need a global array in the scripts to handle the service name and the cleanup steps. A teardown function may make the code more easy to understand and also make sure the services we start will be cleanup and removed from the test machine. Thanks a lot for the advice.